src/shared_modules/rsync/src/rsyncImplementation.cpp

```diff
@@ -150,21 +152,34 @@ void RSyncImplementation::registerSyncId(const RSYNC_HANDLE handle,
 
     const auto registerCallback
     {
-        [spDBSyncWrapper, syncConfiguration, callbackWrapper] (const SyncInputData & syncData)
+        [spDBSyncWrapper, syncConfiguration, callbackWrapper, handle] (const SyncInputData & syncData)
         {
             try
             {
-                if (0 == syncData.command.compare("checksum_fail"))
+                if (syncData.id < SynchronizationController::instance().get(handle))
                 {
-                    sendChecksumFail(spDBSyncWrapper, syncConfiguration, callbackWrapper, syncData);
+                    SynchronizationController::instance().set(handle, syncData.id);
```

Comment:
You've already mentioned this algorithm was verified and tested in FIM
https://github.com/wazuh/wazuh/blob/4.5/src/syscheckd/fim_sync.c#L371-L374

But I have a personal question about how this works:
how can a previous sync end successfully if it began the checksum search with an older state **A** but after a new scan, now the agent DB has changed to a newer state **B**? Wouldn't the binary search be calculating left/right checksums over different data?